e2e/framework/testserver.go

```diff
@@ -550,6 +563,171 @@ type PreparedTransaction struct {
 	Input    []byte
 }
 
+type Txn struct {
+	key     *wallet.Key
+	client  *jsonrpc.Eth
+	hash    *ethgo.Hash
+	receipt *ethgo.Receipt
+	raw     *ethgo.Transaction
+	chainID *big.Int
+
+	sendErr error
+	waitErr error
+}
+
+func (t *Txn) Deploy(input []byte) *Txn {
+	t.raw.Input = input
+
+	return t
+}
+
+func (t *Txn) Transfer(to ethgo.Address, value *big.Int) *Txn {
+	t.raw.To = &to
+	t.raw.Value = value
+
+	return t
+}
+
+func (t *Txn) Value(value *big.Int) *Txn {
+	t.raw.Value = value
+
+	return t
+}
+
+func (t *Txn) To(to ethgo.Address) *Txn {
+	t.raw.To = &to
+
+	return t
+}
+
+func (t *Txn) GasLimit(gas uint64) *Txn {
+	t.raw.Gas = gas
+
+	return t
+}
+
+func (t *Txn) GasPrice(price uint64) *Txn {
+	t.raw.GasPrice = price
+
+	return t
+}
+
+func (t *Txn) Nonce(nonce uint64) *Txn {
+	t.raw.Nonce = nonce
+
+	return t
+}
+
+func (t *Txn) sendImpl() error {
+	// populate default values
+	t.raw.Gas = 1048576
+	t.raw.GasPrice = 1048576
```

Comment:
Why not set default value only if they are zero?

Reply:
I did because in the current test that behavior is not used (i.e custom Gas, GasPrice).